### DISCUSSION TITLE
ET calculation for single band raster without time dimension

### DISCUSSION THREAD

**Discussion by hseybold** [pyet-org/pyet#79] [Q&A]
Hi,
I have single layer geotiff rasters  for which I want to calculate ET  
I can read the data into a 2 dimensional xarray. 
However pyet complains about a missing time dimension

I've tried to add a time dimension with a single date to the xarray but was not successful in getting pyet calculating ET (see example below)

Any hint how to calculate 2d ET maps with  pyet from single band rasters is appreciated.
Thank you very much

Example 

pet_oudin = pyet.oudin(tmean_ds, lat=lat), where tmean_ds and lat are loaded from single band geotiffs using

tmean = rioxarray.open_rasterio("./normals/PRISM_tmean_30yr_normal_800mM5_annual_bil.tif")
lat   = rioxarray.open_rasterio("Lat.tif")

**Comment by mvremec** [comment]
Hi @hseybold,

Glad to hear you are trying to use PyET!
Based on what you wrote, I imagine you want to compute PET using a 30-year average tmean.

The Oudin method and PyET are not directly designed to do that as they are designed to work with daily values. For example, the Oudin equation internally computes the daily extraterrestrial radiation, which is computed based on the Day of the Year, which it gets from the time dimension of tmean, and latitude. 

You can:
1. Use a daily time step and xarray.DataArray format with a time dimension of tmean.
or a workaround with the data you have:
2. Compute the mean daily extraterrestrial radiaton (Ra) for this period and then compute PET as:
* Ensure latitude is in radians.

tindex = pd.date_range("2000-1-1", "2000-12-31")  # Create the time dimension
ra = pyet.extraterrestrial_r(tindex, lat).mean()  # Compute the annual mean Ra
pet = ra * (tmean + 5) / 2.45 / 100
pet = pet.where((tmean + 5) >= 0, 0)

Hope this helps!

Best,
Matevz

**Reply by hseybold** [reply]
Hi, 
thanks a lot for the reply. I fell into the trap that Oudin internally requires a DOY calculation. My intention for using Oudin was mainly to test how to pass raster variables  pyet functions using a function with the least number of variables.
If I interpret your response correctly the error only appears for ET equations which require time (DOY) in some way for e.g. estimating solar radiation internally. Thus when using geotiff of a single day I have to add a time dimension with a single datetime or stack the geotiffs in an xarray in these cases?  
In the end the error prevents the miss-use of the function, which is good, but a  pyet specific error message may be helpful when no (correct) datetime dimension is provided and the function requires such an input implicitly.

Thanks a lot for your reply and providing such a useful tool.

Best,
   hansjoerg
